mantid-development/readme.md

```diff
@@ -37,25 +37,37 @@ mounted to locations on the host filesystem. Reasons being:
 - Running a container built or packaged Mantid on the host (assuming an
   appropriate host system and base image)
 
-Of course Docker volumes could also be used for the build and external data
-directories if you will only ever do container builds.
+To ensure file permissions are handled correctly if mapping volumes to your host
+filesystem you must pass the `PUID` and `PGID` environment variables when
+starting a container, these should be set to your user ID and group ID
+respectively.
 
 The container can be run like this (replacing `centos7` with the base OS
 you care about):
 
 ```sh
-docker run --rm -it \
-  -v /path/to/mantid/source:/mantid_src \
-  -v /path/to/mantid/build:/mantid_build \
-  -v /path/to/mantid/data:/mantid_data \
+docker run \
+  --rm \
+  --interactive \
+  --tty \
+  --env PUID=`id -u` \
+  --env PGID=`id -g` \
+  --volume /path/to/mantid/source:/mantid_src \
+  --volume /path/to/mantid/build:/mantid_build \
+  --volume /path/to/mantid/data:/mantid_data \
   mantidproject/mantid-development-ubuntuxenial
 ```
 
 This will give you a `bash` shell in the build directory. From here you can run
 `cmake` and your build tool of choice just as you would on your host OS.
 
+Inside the container you will have the username `abc` and no root/superuser
```

Comment:
Would you consider giving `abc` `sudo` access? I understand not wanting to change what is installed but at least being able to try out new packages or edits that require root quickly could be advantageous rather having to go through a constant image rebuild cycle.

Reply:
Yes, that shouldn't be an issue.

I'll revoke `sudo` in the Dockerfiles for CI machines however, IIRC they should not need full access (other than for system tests IIRC?) and providing it is a generally bad idea.

Reply:
Yep, sounds good.